src/python/tests/assembly/test_download.py

```diff
@@ -0,0 +1,458 @@
+            id="Bad md5 file - tab separated",
+        ),
+        pytest.param(Path("dir_name"), None, does_not_raise(), id="Dir passed instead of file"),
+    ],
+)
+def test_checksums(
+    data_dir: Path, checksum_file: Optional[Path], checksum: str, expectation: ContextManager
+) -> None:
+    """Tests the `download.get_checksums()` function.
+
+    Args:
+        data_dir: Path to test data root dir
+        checksum_file: File name containing checksums
+        checksum: Test MD5 checksum
+        expectation: Context manager expected raise exception
+    """
+    with expectation:
+        md5_input_path = data_dir / checksum_file
+        obtained_checksums = get_checksums(md5_input_path)
+        assert obtained_checksums.get("annotation_hashes.txt", None) == checksum
+
+
+#################
+@pytest.mark.dependency(name="test_md5_files", depends=["test_checksums"])
+@pytest.mark.parametrize(
+    "md5_file, md5_path, checksum_bool",
+    [
+        pytest.param("md5checksums.txt", None, True, id="Normal case"),
+        pytest.param("wrong_md5_checksums.txt", None, False, id="Incorrect md5 checksum"),
+        pytest.param(None, None, True, id="No md5file specified, resort to default"),
+        pytest.param(None, Path("*"), False, id="Incompatible os path '*'"),
+        pytest.param("missingfile_md5.txt", None, False, id="md5 checksum with ref of missing file"),
+    ],
+)
+def test_md5_files(data_dir: Path, md5_file: str, md5_path: Optional[Path], checksum_bool: bool) -> None:
+    """Tests the md5_files() function
+    Args:
+        data_dir: Path to test data root dir
+        md5_file: MD5 file used for test
+        md5_path: Path location to MD5 file
+        checksum_bool: Test comparison checksum value
+    """
+    if md5_file is None:
+        return_bool_on_md5files = md5_files(data_dir, md5_path=md5_path)
+    else:
+        return_bool_on_md5files = md5_files(data_dir, md5_path=md5_path, md5_filename=md5_file)
+    assert return_bool_on_md5files == checksum_bool
+
+
+#################
+@pytest.mark.dependency(name="test_download_single_file")
+@pytest.mark.parametrize(
+    "ftp_file, md5_sums, expectation",
+    [
+        pytest.param(
+            "test_ftp_file.txt",
+            {"test_ftp_file.txt": "e98b980b442fdb2a21877dcc55e11848"},
+            does_not_raise(),
+            id="Download OK file",
+        ),
+        pytest.param(
+            "non-existing-file.txt",
+            {"test_ftp_file.txt": "e98b980b442fdb2a21877dcc55e11848"},
+            pytest.raises(FileNotFoundError),
+            id="Can't download non-existing file",
+        ),
+        pytest.param(
+            "non-existing-file.txt",
+            {"non-existing-file.txt": "e98b980b442fdb2a21877dcc55e11848"},
+            pytest.raises(ftp_error_reply),
+            id="File has md5sum but does not exist",
+        ),
+        pytest.param(
+            "test_ftp_file.txt",
+            {"test_ftp_file.txt": ""},
```

Comment:
```suggestion
            {"test_ftp_file.txt": "e98b980b442fdb2a21877dcc55e11848"},
```
Missing checksum to force retry.